app/vmauth/auth_config.go

```diff
@@ -693,6 +693,12 @@ func loadAuthConfig() (bool, error) {
 	authConfig.Store(ac)
 	authConfigData.Store(&data)
 	authUsers.Store(&m)
+	if prevAc != nil {
```

Comment:
Can we move it to L:690? Should it be a part of `UnregisterSet` function?

Reply:
It may result in metric gaps.

AtomicValue.Store function is fast and has low probability of potential missing metrics during scrape. While UnregisterAllMetrics could be slow and vmagent (or any other scraping tool) could send stale markers for missing metrics.

I'm not sure, that `UnregisterSet` is a great place for it.

 Maybe a good option for it, move out global state from metrics package to local state inside `Set`.